src/app/learnocaml_teacher_tab.ml

```diff
@@ -299,13 +299,11 @@ let rec teacher_tab token _select _params () =
               H.td [stars_div meta.Exercise.Meta.stars];
               H.td [
                 let cls, text =
-                  match Token.Map.is_empty ES.(st.assignments.token_map),
-                        ES.(st.assignments.default) with
-                  | true, ES.Open -> "exo_open", [%i"Open"]
-                  | true, ES.Closed -> "exo_closed", [%i"Closed"]
-                  | _, (ES.Assigned _ | ES.Closed) ->
-                      "exo_assigned", [%i"Assigned"]
-                  | false, ES.Open -> "exo_assigned", [%i"Open/Assg"]
+                  match ES.(is_open_assg_globally st.assignments) with
+                  | `Open -> "exo_open", [%i"Open"]
+                  | `OpenAssg -> "exo_assigned", [%i"Open/Assg"]
+                  | `Assigned -> "exo_assigned", [%i"Assigned"]
+                  | `Closed -> "exo_closed", [%i"Closed"]
```

Comment:
Nitpicking: I am not a big fan of writing "Assg" for "Assignment"/"Assigned".

Reply:
Maybe `Open/Assigned`